pipeline_dp/mean_var_params.py

```diff
@@ -0,0 +1,117 @@
+"""DP computing of count, sum, mean, variance."""
+
+import numpy as np
+
+from dataclasses import dataclass
+from pipeline_dp.aggregate_params import NoiseKind
+
+
+@dataclass
+class MeanVarParams:
+	"""Manages the parameters used for computing the mean/variance."""
+	eps: float
+	delta: float
+	low: float
+	high: float
+	sensitivity: float  # L1 for Laplace, L2 for Gaussian
+	noise_kind: NoiseKind  # Laplace or Gaussian
+
+
+def _apply_laplace_mechanism(value, eps, sensitivity):
+	"""Applies the Laplace mechanism.
+
+	Args:
+		value: The result of querying the database.
+		eps: Parameter of (epsilon, delta)-differential privacy.
+		sensitivity: The L1-sensitivity of the query.
+	"""
+	return value + np.random.laplace(0, sensitivity / eps)
+
+
+def _apply_gaussian_mechanism(value, eps, delta, sensitivity):
+	"""Applies the Gaussian mechanism.
+
+	Args:
+		value: The result of querying the database.
+		eps, delta: Parameters of (epsilon, delta)-differential privacy.
+		sensitivity: The L2-sensitivity of the query.
+	"""
+	return value + np.random.normal(0, np.var(eps, delta, sensitivity))
```

Comment:
is np.var supposed to be a function that computes variance of the noise by the formula from the "DP book??

Reply:
Actually no, I replaced it with the formula from the book.